report-ng/app/src/components/timings/test-timings/test-timings.ts

```diff
@@ -0,0 +1,357 @@
+        if (params.config) {
+            this._showConfigurationMethods = !!params.config.toLowerCase();
+        }
+        this._getLookUpOptions()
+        this._filter();
+    }
+
+    selectedOptionIdChanged(methodId: string) {
+        if (methodId) {
+            this._methodId = methodId;
+            this.queryParams.methodId = methodId
+            this.updateUrl(this.queryParams);
+            this._highlightData()
+        }
+    }
+
+    selectionChanged() { // called when input from lookup text field is deleted manually
+        this._setChartOption(); // overwrites color highlighting (reset)
+        if (this._inputValue.length == 0) {
+            this._methodId = undefined;
+            delete this.queryParams.methodId;
+            this.updateUrl({rangeNum: this.queryParams.rangeNum, methodId: this._methodId});
+            this._getLookUpOptions();
+        }
+    }
+
+    private _getLookUpOptions() {
+        this._statisticsGenerator.getExecutionStatistics().then(executionStatistics => {
+            let methodContexts: IMethodContext[];
+            if (this.queryParams.methodId) {
+                methodContexts = [executionStatistics.executionAggregate.methodContexts[this.queryParams.methodId]];
+                this._highlightData();
+                this.updateUrl({methodId: this.queryParams.methodId});
+                this._methodId = this.queryParams.methodId;
+            } else if (this._inputValue?.length > 0) {
+                this._searchRegexp = this._statusConverter.createRegexpFromSearchString(this._inputValue);
+                delete this.queryParams.methodId;
+                methodContexts = Object.values(executionStatistics.executionAggregate.methodContexts)
+                    .filter(methodContext => methodContext.contextValues.name.match(this._searchRegexp))
+            } else {
+                methodContexts = Object.values(executionStatistics.executionAggregate.methodContexts);
+            }
+            if (!this._showConfigurationMethods) {
+                methodContexts = methodContexts.filter(methodContext => methodContext.methodType == MethodType.TEST_METHOD);
+            }
+            this._lookUpOptions = methodContexts.map(methodContext => methodContext.contextValues).sort(function (a, b) {
+                if (a.name < b.name) {
+                    return -1;
+                }
+                if (a.name > b.name) {
+                    return 1;
+                }
+                return 0;
+            });
+        });
+    };
+
+    private _filter() {
+        this._loading = true;
+
+        this._methodDetails = [];
+
+        this._statisticsGenerator.getExecutionStatistics().then(executionStatistics => {
+            this._executionStatistics = executionStatistics;
+
+            executionStatistics.classStatistics
+                .flatMap(classStatistic => {
+                    this._methodDetails = executionStatistics.classStatistics
```

Comment:
Please see note above. If you struggle unterstanding `flatMap` then please ask @martingrossmann